heaps/min_heap.py

```diff
@@ -57,18 +70,40 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: O(log n)
+            Space complexity: O(1)
         """
-        pass
+        temporary = index
+        while temporary > 0:
+            parent = (temporary - 1) // 2
+            if self.store[temporary].key < self.store[parent].key:
+                self.swap(temporary, parent)
+                temporary = parent
+            else:
+                break
```

Comment:
Consider reversing the sense of the comparison to treat it more like a guard clause, which helps avoid the dangling else and lets us unindent the main logic.

```suggestion
            if self.store[temporary].key >= self.store[parent].key:
                break

            self.swap(temporary, parent)
            temporary = parent
```